fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1119,15 +1119,31 @@ public static void stopOngoingOnlineIndexBuilds(@Nonnull FDBRecordStore recordSt
         SynchronizedSession.endAnySession(recordStore.ensureContextActive(), indexBuildLockSubspace(recordStore, index));
     }
 
-    @VisibleForTesting
-    CompletableFuture<Void> checkNoOngoingOnlineIndexBuildsAsync() {
-        return runner
-                .runAsync(context -> openRecordStore(context).thenApply(store -> indexBuildLockSubspace(store, index)))
-                .thenCompose(lockSubspace ->
-                        // It will throw {@link com.apple.foundationdb.synchronizedsession.SynchronizedSessionLockedException}
-                        // if there are ongoing online index builds.
-                        runner.startSynchronizedSessionAsync(lockSubspace, leaseLengthMills))
-                .thenCompose(SynchronizedSessionRunner::endSessionAsync);
+    /**
+     * Synchronous/blocking version of {@link #checkAnyOngoingOnlineIndexBuildsAsync()}.
+     * @return <code>true</code> if the index is being built and <code>false</code> otherwise
+     */
+    public boolean checkAnyOngoingOnlineIndexBuilds() {
+        return runner.asyncToSync(FDBStoreTimer.Waits.WAIT_CHECK_ONGOING_ONLINE_INDEX_BUILD, checkAnyOngoingOnlineIndexBuildsAsync());
+    }
+
+    /**
+     * Check if the index is being built by any of the {@link OnlineIndexer}s (only if they use {@link SynchronizedSession}s).
+     * @return a future that will complete to <code>true</code> if the index is being built and <code>false</code> otherwise
+     */
+    public CompletableFuture<Boolean> checkAnyOngoingOnlineIndexBuildsAsync() {
+        return runner.runAsync(context -> openRecordStore(context).thenCompose(recordStore ->
+                checkAnyOngoingOnlineIndexBuildsAsync(recordStore, index)));
+    }
+
+    /**
+     * Check if the index is being built by any of the {@link OnlineIndexer}s (only if they use {@link SynchronizedSession}s).
```

Comment:
Same comments about the docs as on the other method

Reply:
Removed `the`, but I don't I should call out "self conflict" here because this is a static method.